diff --git a/node_modules/@swc/core/Visitor.js b/node_modules/@swc/core/Visitor.js
index 9d359a3..d50974d 100644
--- a/node_modules/@swc/core/Visitor.js
+++ b/node_modules/@swc/core/Visitor.js
@@ -1240,7 +1240,7 @@ class Visitor {
         return n;
     }
     visitObjectPattern(n) {
-        n.props = this.visitObjectPatternProperties(n.props);
+        n.props = this.visitObjectPatternProperties(n.props || []);
         n.typeAnnotation = this.visitTsTypeAnnotation(n.typeAnnotation);
         return n;
     }
